src/main/java/bg/libapp/libraryapp/service/BookService.java

```diff
@@ -179,4 +172,104 @@ private String getJsonOfBook(Book book) {
+            specification = BookSpecifications.fieldLike(PUBLISHER, bookFilterRequest.getPublisher()).and(Specification.where(specification));
+        }
+        if (bookFilterRequest.getYearFrom() != null) {
+            specification = BookSpecifications.fieldGreaterThanOrEqual(YEAR, bookFilterRequest.getYearFrom()).and(Specification.where(specification));
+        }
+        if (bookFilterRequest.getYearTo() != null) {
+            specification = BookSpecifications.fieldLowerThanOrEqual(YEAR, bookFilterRequest.getYearTo()).and(Specification.where(specification));
+        }
+        if (bookFilterRequest.getGenres() != null) {
+            for (Integer genre : bookFilterRequest.getGenres()) {
+                specification = BookSpecifications.genreEquals(genre).and(Specification.where(specification));
+            }
+        }
+        if (bookFilterRequest.getAuthorsFirstName() != null) {
+            for (String authorFirstName : bookFilterRequest.getAuthorsFirstName()) {
+                specification = BookSpecifications.authorNameEquals(FIRST_NAME, authorFirstName).and(Specification.where(specification));
+            }
+        }
+        if (bookFilterRequest.getAuthorsLastName() != null) {
+            for (String authorLastName : bookFilterRequest.getAuthorsLastName()) {
+                specification = BookSpecifications.authorNameEquals(LAST_NAME, authorLastName).and(Specification.where(specification));
+            }
+        }
+        return specification;
+    }
+
+    public BookDTO changeStatus(String isbn, BookChangeStatusRequest bookChangeStatusRequest) {
+        logger.info("changeStatus of Book method called with params: " + bookChangeStatusRequest);
+
+        Book book = getBookByIsbnOrThrowException(isbn);
+        boolean statusToSet = bookChangeStatusRequest.isActive();
+        if (statusToSet) {
+            activateBook(book);
+        } else {
+            deactivateBook(book, bookChangeStatusRequest.getDeactivateReason());
+        }
+        return BookMapper.mapToBookDTO(book);
+    }
+
+    private void deactivateBook(Book book, String deactivateReason) {
+        if (isValidDeactivateReason(deactivateReason)) {
+            String oldStatus = String.valueOf(book.isActive());
+            String oldReason = book.getDeactivateReason();
+            book.setActive(false);
+            book.setDeactivateReason(deactivateReason);
+            eventPublisher.publishEvent(new UpdateActiveStatusBookAuditEvent(book, oldStatus));
+            eventPublisher.publishEvent(new UpdateDeactivateReasonBookAuditEvent(book, oldReason));
+            logger.info("Deactivated book with isbn '" + book.getIsbn() + "' and params: " + book);
+        } else {
+            logger.error("There is no such a deactivate reason '" + deactivateReason + "'");
+            throw new NoSuchDeactivateReasonException(deactivateReason);
+        }
+    }
+
+    private void activateBook(Book book) {
```

Comment:
Could just do the logic inside the public service method to avoid code duplication (Two 10-line methods have 6 duplicated lines)